R/get_taxonlijsten_items.R

```diff
@@ -0,0 +1,152 @@
+#' to the output. For example, if the originally published taxon on a taxonlist
+#' is 'Cicindela spec.', the interpretation will exist of all relevant
+#' Cicindela species
+#' @param collect If FALSE (the default), a remote tbl object is returned. This
+#' is like a reference to the result of the query but the full result of the
+#' query is not brought into memory. If TRUE the full result of the query is
+#' collected (fetched) from the database and brought into memory of the working
+#' environment.
+#'
+#' @return A remote tbl object (collect = FALSE) or a tibble dataframe (collect
+#' = TRUE) with variables Lijst, Publicatiejaar, LaatsteVersie, Taxongroep,
+#' Naamwet_interpretatie, Auteur, NaamNed_interpretatie, Kenmerk,
+#' KenmerkwaardeCode, Kenmerkwaarde and extra variables Taxongroep_origineel,
+#' Naamwet_origineel, Naamned_origineel when requested (original = TRUE)
+#'
+#' @importFrom glue glue_sql
+#' @importFrom DBI dbGetQuery
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr collect tbl sql select
+#'
+#' @export
+#' @family taxonlijsten
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' library(tidyverse)
+#' con <- connect_inbo_dbase("D0156_00_Taxonlijsten")
+#'
+#' # Get all taxa from list 'Jachtdecreet'
+#' get_taxonlijsten_items(con, list =  'Jachtdecreet', collect = TRUE)
+#'
+#' # Get all taxa on category 2 of 'Soortenbesluit'
+#' get_taxonlijsten_items(con, list =  'soortenbesluit', feature = 'cat2')
+#'
+#' # Get all taxonlist that include 'Gentiaanblauwtje'
+#' get_taxonlijsten_items(con, taxon = 'Gentiaanblauwtje', collect = TRUE)
+#'
+#' # Get all taxa with red list status CR (critically endangered)
+#' get_taxonlijsten_items(con, feature = 'CR')
+#'
+#' # Get original and interpreted Cicindela taxa from list 'Soortenbesluit'
+#' get_taxonlijsten_items(con, list = 'Soortenbesluit', taxon = '%Cicindela%'
+#' , original = TRUE) %>%
+#' select(c('Naamwet_origineel', 'NaamNed_origineel', 'Naamwet_interpretatie'
+#' , 'NaamNed_interpretatie'))
+#'
+#' # Compare red list status on multiple listversions
+#' get_taxonlijsten_items(con, version = 'all'
+#' , list = 'rode lijst van de dagvlinders') %>%
+#' select(c('Lijst', 'Publicatiejaar', 'Naamwet_interpretatie'
+#' , 'NaamNed_interpretatie', 'KenmerkwaardeCode')) %>%
+#' pivot_wider(names_from = Publicatiejaar, values_from = KenmerkwaardeCode)
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
+#' }
+
+get_taxonlijsten_items <- function(connection,
+                                       list = "%",
+                                       taxon = "%",
+                                       feature = "%",
+                                       version = c("latest", "old", "all"),
+                                       original = FALSE,
+                                       collect = FALSE
+) {
+
+  assert_that(is.character(list))
+
+  assert_that(is.character(taxon))
+
+  assert_that(is.character(feature))
+
```

Comment:
```suggestion

  assert_that(is.logical(original))

```

Just a suggestion to wake up unattent users.  (I tried `original <- "ja"` and `"isTRUE(original)"`. ;-) )  I usually try to cover all arguments with a test to avoid unwanted outcome or unclear error messages (and questions from users), but I leave it up to you to decide how far you want to go with it.